### ISSUE TITLE
User and group ids in /etc/subuid and /etc/subgid are not used in the order specified

### ISSUE THREAD

**Issue by dohse** [moby/moby#39301]
**Description**

Moby currently sorts uid and gid ranges in id maps. This causes subuid and subgid files to be interpreted wrongly.

**Steps to reproduce the issue:**
1. Enable `userns-remap` for docker.
2. Configure a subuid file which does not specify the host user id in ascending order. For instance:
    
```
> cat /etc/subuid
jonas:100000:1000
jonas:1000:1
```

3. Restart docker
4. Check resulting host to container user namespace mapping with `docker run ubuntu cat /proc/self/uid_map`

**Describe the results you received:**

Host user id 1000 is mapped to container user id 0 (root) and host user ids 100000-100999 are mapped to container user ids 1-1000.
```
> docker run ubuntu cat /proc/self/uid_map
         0       1000          1
         1     100000       1000
```

**Describe the results you expected:**

Host user id 1000 is mapped to container user id 1000 and host user ids 100000-100999 are mapped to container user ids 0-999.

```
> docker run ubuntu cat /proc/self/uid_map
         0     100000       1000
      1000       1000          1
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
> docker version
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:35:57 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 1
Server Version: 18.09.6
Storage Driver: btrfs
 Build Version: Btrfs v4.7.3
 Library Version: 101
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: bb71b10fd8f58240ca47fbb579b9d1028eea7c84
runc version: 2b18fe1d885ee5083ef9f0838fee39b62d653e30
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
 userns
Kernel Version: 4.15.0-50-generic
Operating System: Ubuntu 18.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.43GiB
Name: Minun
ID: L3XT:OUUV:BHHA:S2QW:W3WN:TTHD:62AZ:K5BI:X4FJ:XW6R:ZQVD:DX6P
Docker Root Dir: /var/lib/docker/1000.999
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by dohse** [comment]
I created a pull request that should fix this issue: https://github.com/moby/moby/pull/39288

**Comment by yongtang** [comment]
Close as the issue seems to have been fixed in PR #39288
